
<div style="margin-left: 10px">
<h1>Gol Linhas Aéreas Inteligentes - Case 02</h1>
Processo seletivo - Desenvolvedor (E)<br>
<b>Candidato:</b> Maycon Felipe da Silva Mota <br>
<b>Email:</b> mayconfelipemotamw3@gmail.com<br>


---





# Importando bibliotecas que usarei no CASE 02

In [1]:
import pandas as pd 
import numpy as np 
import seaborn as sns
import dataframe_image as dfi

<p> Para a realização do case 02 irei importar o conjunto de dados disponibilizados pela ANAC que se refere aos microdados estatísticos referente ao transporte aéreo em Maio de 2021 enviado pelas companhias áereas a Agência Nacional de Aviação Cívil </p>

In [2]:
# Realizar leitura do dataset básica disponível pela ANAC em https://www.gov.br/anac/pt-br/assuntos/regulados/empresas-aereas/envio-de-informacoes/microdados/microdados

df_case02 = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';')

# Verificar os tipos dos campos 
df_case02.dtypes

C:\Users\mayco\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


id_basica            int64
id_empresa           int64
sg_empresa_icao     object
sg_empresa_iata     object
nm_empresa          object
                    ...   
nr_rtk               int64
id_arquivo           int64
nm_arquivo          object
nr_linha           float64
dt_sistema          object
Length: 110, dtype: object

In [3]:
# Analisando como está a distribuição das variáveis númericas do conjunto de dados

df_case02.describe().transpose()

count          mean           std         min  \
id_basica                  36100.0  2.157449e+07  1.044343e+04  21556193.0   
id_empresa                 36100.0  1.000936e+06  3.023686e+02   1000002.0   
nr_voo                     36100.0  4.159456e+03  1.837294e+03        11.0   
id_di                      36100.0  2.399391e+00  1.628709e+00         2.0   
nr_ano_referencia          36100.0  2.021000e+03  0.000000e+00      2021.0   
nr_semestre_referencia     36100.0  1.000000e+00  0.000000e+00         1.0   
nr_trimestre_referencia    36100.0  2.000000e+00  0.000000e+00         2.0   
nr_mes_referencia          36100.0  5.000000e+00  0.000000e+00         5.0   
nr_semana_referencia       36100.0  2.067039e+01  1.297244e+00        18.0   
nr_dia_referencia          36100.0  1.658958e+01  8.839623e+00         1.0   
nr_ano_mes_referencia      36100.0  2.021050e+05  0.000000e+00    202105.0   
id_tipo_linha              36100.0  9.982825e-01  5.149875e-01         0.0   
nr_ano_partida_real        36100.0  2.021000e+03  0.000000e+00      2021.0   
nr_semestre_partida_real   36100.0  1.000000e+00  0.000000e+00         1.0   
nr_trimestre_partida_real  36100.0  2.000000e+00  0.000000e+00         2.0   
nr_mes_partida_real        36100.0  5.000332e+00  2.232752e-02         4.0   
nr_semana_partida_real     36100.0  2.067238e+01  1.297365e+00        18.0   
nr_dia_partida_real        36100.0  1.659557e+01  8.839269e+00         1.0   
nr_ano_mes_partida_real    36100.0  2.021050e+05  2.232752e-02    202104.0   
id_aerodromo_origem        36100.0  5.573086e+02  1.690043e+03         1.0   
nr_etapa                   36100.0  1.074958e+00  3.378798e-01         1.0   
nr_ano_chegada_real        36099.0  2.021000e+03  0.000000e+00      2021.0   
nr_semestre_chegada_real   36099.0  1.000000e+00  0.000000e+00         1.0   
nr_trimestre_chegada_real  36099.0  2.000000e+00  0.000000e+00         2.0   
nr_mes_chegada_real        36099.0  5.003574e+00  5.967273e-02         5.0   
nr_semana_chegada_real     36099.0  2.068628e+01  1.295629e+00        18.0   
nr_dia_chegada_real        36099.0  1.659650e+01  8.843347e+00         1.0   
nr_ano_mes_chegada_real    36099.0  2.021050e+05  5.967273e-02    202105.0   
id_equipamento             36100.0  2.061002e+02  1.951714e+02         5.0   
id_aerodromo_destino       36100.0  5.598016e+02  1.703600e+03         1.0   
nr_escala_destino              0.0           NaN           NaN         NaN   
lt_combustivel             36100.0  4.840496e+03  7.584977e+03         0.0   
nr_assentos_ofertados      36100.0  1.416720e+02  7.222836e+01         0.0   
kg_payload                 36100.0  2.040435e+04  1.706992e+04         0.0   
km_distancia               36100.0  1.543563e+03  2.035418e+03         0.0   
nr_passag_pagos            36100.0  1.066604e+02  5.892433e+01         0.0   
nr_passag_gratis           36100.0  2.612825e+00  3.419583e+00         0.0   
kg_bagagem_livre           36100.0  6.316227e+02  5.290116e+02         0.0   
kg_bagagem_excesso         36100.0  1.299562e+01  2.579348e+01         0.0   
kg_carga_paga              36100.0  3.634506e+03  1.183644e+04         0.0   
kg_carga_gratis            36100.0  1.597410e+01  9.986982e+01         0.0   
kg_correio                 36100.0  1.865680e+02  1.342465e+03         0.0   
nr_decolagem               36100.0  9.979224e-01  4.553354e-02         0.0   
kg_peso                    36100.0  1.274575e+04  1.113539e+04         0.0   
nr_pax_gratis_km           36100.0  3.449034e+03  1.119979e+04         0.0   
nr_carga_paga_km           36100.0  1.942904e+07  7.901612e+07         0.0   
nr_carga_gratis_km         36100.0  3.488502e+04  4.133892e+05         0.0   
nr_correio_km              36100.0  3.250681e+05  2.494234e+06         0.0   
nr_bagagem_paga_km         36100.0  1.471909e+04  4.910776e+04         0.0   
nr_bagagem_gratis_km       36100.0  8.685774e+05  2.329333e+06         0.0   
nr_ask                     36100.0  2.617207e+05  5

<p> Para a realização do case 02 irei importar o conjunto de dados disponibilizados pela ANAC que se refere aos microdados estatísticos referente ao transporte aéreo em Maio de 2021 enviado pelas companhias áereas a Agência Nacional de Aviação Cívil </p>

In [4]:
# Agrupar por empresa aérea agregando pela quantidade de voos realizados (id_basica)
qtd_voos = df_case02.groupby(by = 'nm_empresa')[['id_basica']].agg(['count'])
qtd_voos.columns = ['Quantidade de voos'] # renomear a coluna
qtd_voos = qtd_voos.sort_values(by = 'Quantidade de voos', ascending=False) # ordenação pelo maior valor ao menor 
qtd_voos.to_csv('./output/qtd_voos_empresa_aerea.csv') # exportar em um CSV
qtd_voos # exibir o dataframe 

Quantidade de voos
nm_empresa                                                            
AZUL LINHAS AÉREAS BRASILEIRAS S/A                               14943
TAM LINHAS AÉREAS S.A.                                            8144
GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S...                6892
SIDERAL LINHAS AÉREAS LTDA.                                        709
PASSAREDO TRANSPORTES AÉREOS S.A.                                  616
...                                                                ...
ETIHAD AIRWAYS P.J.S.C.                                              3
AIRES - AEROVÍAS DE INTEGRACÍON REGIONAL S.A.                        2
SURINAM AIRWAYS LTD                                                  2
AMERICAN JET S.A.                                                    2
AEROVIAS DEL CONTINENTE AMERICANO S.A. AVIANCA                       2

[65 rows x 1 columns]

<p>Em Maio, a retomada da aviação foi liderada pela Azul Linhas Aéreas, com mais de 14.943 voos realizados, seguida pela LATAM Linhas Aéreas e em terceiro: a Gol Linhas Aéreas.</p>

In [25]:
# Agrupar pelo nome da empresa, agregando pela quantidade, média e soma dos indicadores RPK e ASK

qtd_voos = df_case02.query('nm_empresa == "AZUL LINHAS AÉREAS BRASILEIRAS S/A" | nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)" | nm_empresa == "TAM LINHAS AÉREAS S.A."').groupby(by = 'nm_empresa')[['nr_rpk', 'nr_ask']].agg(['count', 'sum', 'mean'])
# qtd_voos.columns = ['Quantidade de voos']
# qtd_voos = qtd_voos.sort_values(by = 'Quantidade de voos', ascending=False)

qtd_voos.sort_values(by = [('nr_rpk', 'sum')], ascending=False) ## ordenação pelo maior valor ao menor 
qtd_voos.query('nm_empresa == "AZUL LINHAS AÉREAS BRASILEIRAS S/A" | nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)" | nm_empresa == "TAM LINHAS AÉREAS S.A."').to_csv('./output/nrrpk_nrask_por_empresa_aerea.csv') ## Filtrar apenas entre GOL,
qtd_voos.to_csv('./output/soma_media_nrrpk_nrask_azul_gol.csv') # exportar em um CSV

qtd_voos # Exibir o dataframe
 

nr_rpk              \
                                                    count         sum   
nm_empresa                                                              
AZUL LINHAS AÉREAS BRASILEIRAS S/A                  14943  1531705713   
GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S...   6892  1219640321   
TAM LINHAS AÉREAS S.A.                               8144  1445463066   

                                                                  nr_ask  \
                                                             mean  count   
nm_empresa                                                                 
AZUL LINHAS AÉREAS BRASILEIRAS S/A                  102503.226461  14943   
GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S...  176964.643210   6892   
TAM LINHAS AÉREAS S.A.                              177488.097495   8144   

                                                                               
                                                           sum           mean  
nm_empresa                                                                     
AZUL LINHAS AÉREAS BRASILEIRAS S/A                  2020426449  135208.890383  
GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S...  1385225282  200990.319501  
TAM LINHAS AÉREAS S.A.                              2029668313  249222.533522

<p>Analisando os indicadores que referem a demanda do transporte aéreo estabelecidos pela OACI como RPK - métrica que deve mostrar o número de quilômetros percorridos pelos passageiros pagantes, feito pela multiplicação da quantidade de passageiros pela distância total percorrida e ASK - Assentos disponíveis por quilômetros, apesar da Azul ter um maior número de RPK e ASK tendo em vista que seus vôos internacionais, como por exemplo para o continente americano e Europa, o que fez com que a soma dos valores de RPK e ASK as mantivesse a Azul em primeiro lugar, a os valores pela média acabam reposicionando a Gol Linhas Aéreas em segunda posição.</p>

<p> Abaixo, iremos analisar como foi o comportamento envolvendo o TOP 15 voos da Azul Linhas Aéreas em relação a Gol Linhas Aéreas</p>

In [6]:
## Buscar apenas os voos da Azul e GOL 
qtd_voos_origem_destino = df_case02[(df_case02['nm_empresa'] == 'AZUL LINHAS AÉREAS BRASILEIRAS S/A') | (df_case02['nm_empresa'] == 'GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)')]

## Gerar o TOP 15 de voos pela Azul, agregando a quantidade de voos por trecho entre aeroportos
qtd_voos_origem_destino = qtd_voos_origem_destino.groupby(by = ['nm_empresa', 'sg_iata_origem', 'nm_municipio_origem', 'sg_iata_destino', 'nm_municipio_destino']).agg({'count'})

## Gerar a oordenação, renomear e resetar o índice do Dataframe
qtd_voos_origem_destino = qtd_voos_origem_destino['id_basica'].sort_values(by= 'count', ascending=False)

qtd_voos_origem_destino.columns = ['Quantidade de Voos']
qtd_voos_origem_destino = qtd_voos_origem_destino.reset_index()
qtd_voos_origem_destino.query('nm_empresa == "AZUL LINHAS AÉREAS BRASILEIRAS S/A"').head(15).to_csv('./output/top15_voos_realizados_azul.csv')

# Exibir o dataframe
qtd_voos_origem_destino.query('nm_empresa == "AZUL LINHAS AÉREAS BRASILEIRAS S/A"').head(15)

nm_empresa sg_iata_origem   nm_municipio_origem  \
0   AZUL LINHAS AÉREAS BRASILEIRAS S/A            CNF               CONFINS   
1   AZUL LINHAS AÉREAS BRASILEIRAS S/A            VCP              CAMPINAS   
2   AZUL LINHAS AÉREAS BRASILEIRAS S/A            POA          PORTO ALEGRE   
3   AZUL LINHAS AÉREAS BRASILEIRAS S/A            VCP              CAMPINAS   
4   AZUL LINHAS AÉREAS BRASILEIRAS S/A            VCP              CAMPINAS   
5   AZUL LINHAS AÉREAS BRASILEIRAS S/A            SDU        RIO DE JANEIRO   
6   AZUL LINHAS AÉREAS BRASILEIRAS S/A            SDU        RIO DE JANEIRO   
7   AZUL LINHAS AÉREAS BRASILEIRAS S/A            CGH             SÃO PAULO   
8   AZUL LINHAS AÉREAS BRASILEIRAS S/A            VCP              CAMPINAS   
10  AZUL LINHAS AÉREAS BRASILEIRAS S/A            CWB  SÃO JOSÉ DOS PINHAIS   
12  AZUL LINHAS AÉREAS BRASILEIRAS S/A            SDU        RIO DE JANEIRO   
13  AZUL LINHAS AÉREAS BRASILEIRAS S/A            CNF               CONFINS   
14  AZUL LINHAS AÉREAS BRASILEIRAS S/A            VCP              CAMPINAS   
15  AZUL LINHAS AÉREAS BRASILEIRAS S/A            BSB              BRASÍLIA   
16  AZUL LINHAS AÉREAS BRASILEIRAS S/A            MAO                MANAUS   

   sg_iata_destino  nm_municipio_destino  Quantidade de Voos  
0              VCP              CAMPINAS                 224  
1              CNF               CONFINS                 219  
2              VCP              CAMPINAS                 216  
3              POA          PORTO ALEGRE                 215  
4              SDU        RIO DE JANEIRO                 209  
5              VCP              CAMPINAS                 206  
6              CGH             SÃO PAULO                 185  
7              SDU        RIO DE JANEIRO                 183  
8              CWB  SÃO JOSÉ DOS PINHAIS                 168  
10             VCP              CAMPINAS                 168  
12             CNF               CONFINS                 157  
13             SDU        RIO DE JANEIRO                 155  
14             BSB              BRASÍLIA                 150  
15             VCP              CAMPINAS                 148  
16             VCP              CAMPINAS                 146

In [7]:

# Exibir o mesmo dataframe, porém filtrando apenas os dados relativos a GOL Linhas Aéreas S.A e gerar o TOP 15.

qtd_voos_origem_destino.query('nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)"').head(15).to_csv('./output/top15_voos_realizados_gol.csv')
qtd_voos_origem_destino.query('nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)"').head(15)

nm_empresa sg_iata_origem  \
9   GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            SDU   
11  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            CGH   
50  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            GRU   
51  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            BSB   
58  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            CNF   
59  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            REC   
60  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            GRU   
61  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            POA   
62  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            GRU   
63  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            FLN   
64  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            GRU   
65  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            GRU   
67  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            GRU   
69  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            FOR   
70  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...            CWB   

     nm_municipio_origem sg_iata_destino nm_municipio_destino  \
9         RIO DE JANEIRO             CGH            SÃO PAULO   
11             SÃO PAULO             SDU       RIO DE JANEIRO   
50             GUARULHOS             BSB             BRASÍLIA   
51              BRASÍLIA             GRU            GUARULHOS   
58               CONFINS             GRU            GUARULHOS   
59                RECIFE             GRU            GUARULHOS   
60             GUARULHOS             REC               RECIFE   
61          PORTO ALEGRE             GRU            GUARULHOS   
62             GUARULHOS             CNF              CONFINS   
63         FLORIANÓPOLIS             GRU            GUARULHOS   
64             GUARULHOS             FLN        FLORIANÓPOLIS   
65             GUARULHOS             POA         PORTO ALEGRE   
67             GUARULHOS             FOR            FORTALEZA   
69             FORTALEZA             GRU            GUARULHOS   
70  SÃO JOSÉ DOS PINHAIS             GRU            GUARULHOS   

    Quantidade de Voos  
9                  168  
11                 166  
50                 102  
51                 100  
58                  98  
59                  98  
60                  97  
61                  97  
62                  97  
63                  96  
64                  96  
65                  96  
67                  95  
69                  95  
70                  94

<p>Tanto a Azul Linhas Aéreas quanto a Gol linhas Aéreas, utilizam o paradigma spoke-hub de distribuição de voos. O paradigma consiste em centralizar os voos em Aeroportos onde há maior número de voos saindo para atender outras cidades, então, os aeroportos onde a movimentação não é tão expressiva deve ser ligados em um Aeroporto onde há o maior número de voos. O spoke-hub está mais presente na Azul, uma vez que seu HUB principal é VCP. Enquanto na Gol, o HUB é GRU e CGH. Diferente da Gol, a Azul teve um número mais expressivo em trechos comuns entre o HUB ao Aeroporto de Confins (CNF). Isso é devido a Azul ter um expressivo número de voos regionais interligado o interior de Minas Gerais a capital e também ser um HUB estratégico da empresa. </p>  <br> <p>Uma outra análise é do trecho mais comum do Brasil: a ponte aérea. A Azul Linhas Aéreas realizou mais de 185 voos entre Santos Dumont (RJ) para Congonhas (SP), enquanto a Gol realizou 168.  </p>

In [8]:
# Importar o dataset da ANAC, gerar o comparativo entre Azul e GOL entre SDU-CGH e contar a quantidade de voos. 

df_comparativo_sducgh = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';')
df_comparativo_sducgh = df_comparativo_sducgh.query('(nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)" | nm_empresa == "AZUL LINHAS AÉREAS BRASILEIRAS S/A") & sg_iata_origem == "SDU" & sg_iata_destino == "CGH"')
df_comparativo_sducgh.groupby(by= 'nm_empresa')['id_basica'].count()

C:\Users\mayco\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


nm_empresa
AZUL LINHAS AÉREAS BRASILEIRAS S/A                     185
GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)    168
Name: id_basica, dtype: int64

In [9]:
# Importar o dataset da ANAC, buscar apenas os voos da Gol Linhas Aéreas e da Azul entre o trecho SDU-CGH, em seguida, usar a fórmula nr_rpk/nr_ask para entender qual a ocupação entre as duas empresas.

df_comparativo_sducgh = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';') 
df_comparativo_sducgh = df_comparativo_sducgh.query('(nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)" | nm_empresa == "AZUL LINHAS AÉREAS BRASILEIRAS S/A") & sg_iata_origem == "SDU" & sg_iata_destino == "CGH"')
df_comparativo_sducgh['dt_partida_real'] = pd.to_datetime(df_comparativo_sducgh['dt_partida_real'])
df_comparativo_sducgh['diasemana_partida_real'] = df_comparativo_sducgh['dt_partida_real'].dt.day_name(locale='pt')
df_comparativo_sducgh['ocupacao'] = pd.to_numeric(df_comparativo_sducgh['nr_rpk']/df_comparativo_sducgh['nr_ask']*100).astype(float).round(2)
tabela_comparativa = df_comparativo_sducgh.groupby(by= ['nm_empresa'], sort=True)['ocupacao'].describe().transpose()
tabela_comparativa.columns = ['AZUL Linhas Aéreas', 'GOL Linhas Aéreas']
dfi.export(tabela_comparativa,"./output/imgs/df_comparativo_ponteaerea.png")
tabela_comparativa


AZUL Linhas Aéreas  GOL Linhas Aéreas
count          185.000000         166.000000
mean            70.573135          83.191265
std             19.433047          11.131899
min             33.050000          53.450000
25%             52.540000          76.605000
50%             72.030000          85.385000
75%             88.980000          92.385000
max            100.000000          99.440000

Apesar de maior número de voos no trecho SDU-CGH, a Gol Linhas Aéreas apresentou maior média de ocupação, 83.19% de ocupação em seus voos (RPK/ASK), contra 70.57% da Azul Linhas Aéreas.

In [10]:
# Importar o dataset da ANAC, buscar apenas os voos da Gol Linhas Aéreas e da Azul entre o trecho SDU-CGH, em seguida, usar a fórmula nr_rpk/nr_ask para entender qual a ocupação entre as duas empresas.
# Em seguida, entender como está sendo feita a distribuição da ocupação entre os dias da semana entre as duas empresas aéreas

df_comparativo_sducgh = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';')
df_comparativo_sducgh = df_comparativo_sducgh.query('(nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)" | nm_empresa == "AZUL LINHAS AÉREAS BRASILEIRAS S/A") & sg_iata_origem == "SDU" & sg_iata_destino == "CGH"')
df_comparativo_sducgh['dt_partida_real'] = pd.to_datetime(df_comparativo_sducgh['dt_partida_real'])
df_comparativo_sducgh['diasemana_partida_real'] = df_comparativo_sducgh['dt_partida_real'].dt.day_name(locale='pt')
df_comparativo_sducgh['ocupacao'] = pd.to_numeric(round(df_comparativo_sducgh['nr_rpk']/df_comparativo_sducgh['nr_ask']*100, 1))
df_comparativo_sducgh.groupby(by= ['nm_empresa', 'diasemana_partida_real'], sort=True)['ocupacao'].describe().transpose()


C:\Users\mayco\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


nm_empresa             AZUL LINHAS AÉREAS BRASILEIRAS S/A               \
diasemana_partida_real                            Domingo Quarta-feira   
count                                           28.000000    22.000000   
mean                                            89.589286    63.518182   
std                                              7.194151    20.006895   
min                                             69.500000    33.100000   
25%                                             86.875000    45.325000   
50%                                             91.500000    63.150000   
75%                                             94.300000    80.750000   
max                                            100.000000    96.600000   

nm_empresa                                                                \
diasemana_partida_real Quinta-feira Segunda-feira Sexta-feira     Sábado   
count                     28.000000     35.000000   30.000000  19.000000   
mean                      55.503571     82.825714   65.226667  61.057895   
std                       13.450223     13.465388   17.478005  19.815687   
min                       35.600000     52.500000   34.700000  34.700000   
25%                       44.900000     70.300000   47.500000  44.100000   
50%                       52.500000     89.000000   67.350000  57.600000   
75%                       59.550000     91.950000   78.600000  76.250000   
max                       92.400000     97.500000   94.900000  93.200000   

nm_empresa                          \
diasemana_partida_real Terça-feira   
count                    23.000000   
mean                     68.682609   
std                      18.098283   
min                      41.500000   
25%                      54.200000   
50%                      72.000000   
75%                      83.900000   
max                      97.500000   

nm_empresa             GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)  \
diasemana_partida_real                                             Domingo   
count                                                           26.000000    
mean                                                            89.234615    
std                                                              8.939483    
min                                                             61.600000    
25%                                                             86.750000    
50%                                                             91.950000    
75%                                                             94.475000    
max                                                             98.900000    

nm_empresa                                                                  \
diasemana_partida_real Quarta-feira Quinta-feira Segunda-feira Sexta-feira   
count                     19.000000    21.000000     33.000000   28.000000   
mean                      77.078947    83.642857     86.133333   80.110714   
std                       12.089875     9.503345     11.998090   12.122594   
min                       55.200000    58.200000     53.400000   53.800000   
25%                       67.100000    79.700000     82.300000   73.525000   
50%                       80.500000    85.300000     90.600000   81.500000   
75%                       85.750000    91.300000     95.300000   89.925000   
max                       96.600000    96.700000     99.400000   95.400000   

nm_empresa                                     
diasemana_partida_real     Sábado Terça-feira  
count                   21.000000   18.000000  
mean                    79.657143   83.955556  
std                     10.351598    6.858762  
min                     60.100000   69.000000  
25%                     71.300000   80.275000  
50%                     83.700000   84.600000  
75%                     87.600000   88.950000  
max                     93.000000   95.000000

Em uma análise a média de ocupação nos voos entre SDU e CGH é possível identificar que Domingo e Segunda são os dias mais críticos para a operação da ponte aérea para as duas empresas, sendo que a Gol tem obtido maior ocupação na Segunda-Feira, com 86.14, ao contrário da Azul, que teve 82.23%.

In [11]:
# Buscar no Dataset da ANAC, quais trechos da GOL Linhas Aéreas obteve taxa média de ocupação menor que 50%
#  
df_ocupacao_destinos_gol = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';')
df_ocupacao_destinos_gol = df_ocupacao_destinos_gol.query('(nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)")')
df_ocupacao_destinos_gol['dt_partida_real'] = pd.to_datetime(df_ocupacao_destinos_gol['dt_partida_real'])
df_ocupacao_destinos_gol['diasemana_partida_real'] = df_ocupacao_destinos_gol['dt_partida_real'].dt.day_name(locale='pt')
df_ocupacao_destinos_gol['ocupacao'] = df_ocupacao_destinos_gol['nr_rpk']/df_ocupacao_destinos_gol['nr_ask']*100
df_ocupacao_destinos_gol = df_ocupacao_destinos_gol.groupby(by= ['sg_iata_origem', 'nm_municipio_origem', 'sg_iata_destino', 'nm_municipio_destino'], sort=True)[['ocupacao']].mean().sort_values(by= 'ocupacao', ascending=True)
df_ocupacao_destinos_gol

# Exportar quantidade de trechos em uma imagem para anexar na apresentação 
dfi.export(df_ocupacao_destinos_gol[df_ocupacao_destinos_gol.ocupacao < 50],"./output/imgs/df_ocupacao_destinos_gol.png")

# Contar quantos trechos teve menor que 50% de ocupação 
df_ocupacao_destinos_gol[df_ocupacao_destinos_gol.ocupacao < 50].count()

C:\Users\mayco\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


ocupacao    19
dtype: int64

In [12]:
df_ocupacao_destinos_gol[df_ocupacao_destinos_gol.ocupacao < 50] # Exibir o dataframe com trechos e média de ocupação

ocupacao
sg_iata_origem nm_municipio_origem           sg_iata_destino nm_municipio_destino                    
FOR            FORTALEZA                     BVB             BOA VISTA                       0.000000
BAQ            BARRANQUILLA                  GYE             GUAYAQUIL                       0.537634
CKS            PARAUAPEBAS                   SLZ             SÃO LUÍS                       27.150538
SLZ            SÃO LUÍS                      CKS             PARAUAPEBAS                    28.225806
GIG            RIO DE JANEIRO                EZE             EZEIZA, BUENOS AIRES PROVINCE  29.032258
GYE            GUAYAQUIL                     GIG             RIO DE JANEIRO                 29.032258
GIG            RIO DE JANEIRO                BAQ             BARRANQUILLA                   29.569892
EZE            EZEIZA, BUENOS AIRES PROVINCE GIG             RIO DE JANEIRO                 29.569892
GIG            RIO DE JANEIRO                SCL             SANTIAGO                       30.645161
SCL            SANTIAGO                      GIG             RIO DE JANEIRO                 30.645161
ASU            ASUNCIÓN                      CNF             CONFINS                        31.720430
CNF            CONFINS                       ASU             ASUNCIÓN                       31.720430
RBR            RIO BRANCO                    CZS             CRUZEIRO DO SUL                40.903922
VCP            CAMPINAS                      SDU             RIO DE JANEIRO                 43.954451
CNF            CONFINS                       CKS             PARAUAPEBAS                    44.175627
CZS            CRUZEIRO DO SUL               RBR             RIO BRANCO                     44.324600
SDU            RIO DE JANEIRO                VCP             CAMPINAS                       44.985112
CKS            PARAUAPEBAS                   CNF             CONFINS                        45.878136
                                             BSB             BRASÍLIA                       48.757764

Em uma visão macro em relação aos voos executados pela Gol Linhas Aéreas, destaca-se que as piores ocupações foram em novos destinos inaugurados como Paraupebas (27.15% entre CKS-SLZ; 28.22% entre SLZ-CKS) e destinos internacionais como Ezeiza, Santiago, Guayaquil e Barranquilla. Uma hipótese para a baixa ocupação pode ser as restrições e medidas sanitárias de cada destino.

In [13]:
df_ocupacao_destinos_gol[df_ocupacao_destinos_gol.ocupacao > 90].count() # Exibir apenas destinos que tiveram ocupação superior a 90%

ocupacao    81
dtype: int64

In [14]:
df_ocupacao_destinos_gol[df_ocupacao_destinos_gol.ocupacao > 90].sort_values(by= 'ocupacao', ascending=False)  # Ordenar e exibir da maior ocupação a menor

ocupacao
sg_iata_origem nm_municipio_origem     sg_iata_destino nm_municipio_destino           
SDU            RIO DE JANEIRO          CWB             SÃO JOSÉ DOS PINHAIS  98.184044
MAO            MANAUS                  RBR             RIO BRANCO            97.311828
NAT            SÃO GONÇALO DO AMARANTE CNF             CONFINS               97.311828
JDO            JUAZEIRO DO NORTE       GRU             GUARULHOS             96.678187
GRU            GUARULHOS               JDO             JUAZEIRO DO NORTE     96.614894
...                                                                                ...
CNF            CONFINS                 BSB             BRASÍLIA              90.601053
NVT            NAVEGANTES              GIG             RIO DE JANEIRO        90.488007
CWB            SÃO JOSÉ DOS PINHAIS    CNF             CONFINS               90.322581
THE            TERESINA                BSB             BRASÍLIA              90.226718
POA            PORTO ALEGRE            CGH             SÃO PAULO             90.214726

[81 rows x 1 columns]

In [15]:
dfi.export(df_ocupacao_destinos_gol[df_ocupacao_destinos_gol.ocupacao > 90].sort_values(by= 'ocupacao', ascending=False),"./output/imgs/df_ocupacao_destinos_top_gol.png") # Gerar a imagem de ocupação para anexar an apresentação


# Entender a correlação entre horários de voos e soma dos indicadores de RPK e ASK

In [16]:
# Buscar quais são os trechos com maiores números de voos da GOL Linhas Aéreas, em seguida ordenar pelo maior valor ao menor valor 
df_correlacao_horario_voo = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';')
df_correlacao_horario_voo.query('(nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)")').groupby(by= ['nm_empresa', 'sg_icao_origem', 'sg_icao_destino'])[['nr_voo']].count().sort_values(by= 'nr_voo', ascending=False)


C:\Users\mayco\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


nr_voo
nm_empresa                                         sg_icao_origem sg_icao_destino        
GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S... SBRJ           SBSP                168
                                                   SBSP           SBRJ                166
                                                   SBGR           SBBR                102
                                                   SBBR           SBGR                100
                                                   SBCF           SBGR                 98
...                                                                                   ...
                                                   SBPS           SBCF                  1
                                                   SBRB           SBEG                  1
                                                   SBRF           SBJP                  1
                                                   SBSG           SBCF                  1
                                                   UAAA           ZSJN                  1

[259 rows x 1 columns]

In [17]:
## Buscar o voo mais realizado pela GOL para analisar se há correlação entre a faixa horária de partida e a quantidade de passageiros

df_correlacao_horario_voo = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';')

# Filtrar apenas por voos saindo de Guarulhos e chegando em Brasília 
df_correlacao_horario_voo = df_correlacao_horario_voo.query('(nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)") & sg_iata_origem == "GRU" & sg_iata_destino == "BSB"') 

# Filtrar quais colunas serão usadas para a análise
df_correlacao_horario_voo = df_correlacao_horario_voo[['nr_voo', 'sg_iata_origem', 'sg_iata_destino', 'nr_rpk', 'nr_ask', 'dt_partida_real', 'hr_partida_real']]
df_correlacao_horario_voo['hr_partida_real'] = df_correlacao_horario_voo['hr_partida_real'].str.slice(start=0, stop=2)

# Transformar o campo de string para numérico 
df_correlacao_horario_voo['hr_partida_real'] = pd.to_numeric(df_correlacao_horario_voo['hr_partida_real'])

# Exportar o dia da semana 
df_correlacao_horario_voo['dt_semana'] = pd.to_datetime(df_correlacao_horario_voo['dt_partida_real']).dt.day_name(locale='pt')

# Agrupar a quantidade de voos por dia da semana 
df_correlacao_horario_voo.groupby(by= 'dt_semana')[['nr_voo']].count()




C:\Users\mayco\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


nr_voo
dt_semana            
Domingo            16
Quarta-feira       14
Quinta-feira       15
Segunda-feira      18
Sexta-feira        14
Sábado             11
Terça-feira        14

In [18]:
# Buscar apenas os voos da Segunda-Feira, agrupando pela faixa de horário de partida, a quantidade de números de voos 

df_correlacao_horario_voo[['hr_partida_real', 'nr_voo','nr_rpk', 'nr_ask', 'dt_semana']].query('(dt_semana == "Segunda-feira")').groupby(by= 'hr_partida_real')[['nr_voo']].count().reset_index()

hr_partida_real  nr_voo
0                6       2
1                8       5
2               13       2
3               14       3
4               17       2
5               22       4

In [19]:
# Agora, ao invés de buscar a quantidade de voos, buscar a soma de RPK e ASK pela faixa de horário de partida no trecho GRU (Guarulhos) para BSB (Brasília)

df_correlacao_horario_voo = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';')
df_correlacao_horario_voo = df_correlacao_horario_voo.query('(nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)") & sg_iata_origem == "GRU" & sg_iata_destino == "BSB"')
df_correlacao_horario_voo['dt_semana'] = pd.to_datetime(df_correlacao_horario_voo['dt_partida_real']).dt.day_name(locale='pt')
df_correlacao_horario_voo['hr_partida_real'] = df_correlacao_horario_voo['hr_partida_real'].str.slice(start=0, stop=2)

# Realizar o agrupamento através do horário de partida e a agregação pela soma da quantidade de RPK e Número de ASK
df_correlacao_horario_voo = df_correlacao_horario_voo[['hr_partida_real', 'nr_voo','nr_rpk', 'nr_ask', 'dt_semana']].query('(dt_semana == "Segunda-feira")').groupby(by= 'hr_partida_real')[['hr_partida_real', 'nr_rpk', 'nr_ask']].sum().reset_index()

df_correlacao_horario_voo
# df_correlacao_horario_voo = df_correlacao_horario_voo[['hr_partida_real', 'nr_voo','nr_rpk', 'nr_ask', 'dt_semana']].query('(dt_semana == "Segunda-feira")').groupby(by= 'hr_partida_real')[['nr_rpk', 'nr_ask']].sum()

C:\Users\mayco\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


hr_partida_real  nr_rpk  nr_ask
0              06  201780  277020
1              08  639540  795150
2              13  269325  318060
3              14  441180  477090
4              17  269325  318060
5              22  548910  595935

In [20]:
import scipy
print("Teste de normalidade usando Shapiro-Wil para média de RPK: ", scipy.stats.shapiro(df_correlacao_horario_voo['nr_rpk']))
print("Teste de normalidade usando Shapiro-Wil para média de ASK: ", scipy.stats.shapiro(df_correlacao_horario_voo['nr_ask']))

Teste de normalidade usando Shapiro-Wil para média de RPK:  ShapiroResult(statistic=0.9105069637298584, pvalue=0.4397401213645935)
Teste de normalidade usando Shapiro-Wil para média de ASK:  ShapiroResult(statistic=0.8827252984046936, pvalue=0.28180739283561707)


In [21]:
# Verificar se possuem correlação o horário de partida, o número de RPK e ASK.

df_correlacao_horario_voo['hr_partida_real'] = pd.to_numeric(df_correlacao_horario_voo['hr_partida_real'])
df_correlacao_horario_voo
dfi.export(df_correlacao_horario_voo.corr(),"./output/imgs/df_correlacao_soma_ocupacao_gru_bsb.png")
df_correlacao_horario_voo.corr(method='spearman')

hr_partida_real    nr_rpk    nr_ask
hr_partida_real         1.000000  0.318874  0.318874
nr_rpk                  0.318874  1.000000  1.000000
nr_ask                  0.318874  1.000000  1.000000

In [22]:
# Agora, ao invés de buscar a quantidade de voos, buscar a soma de RPK e ASK pela faixa de horário de partida no trecho GRU (Guarulhos) para BSB (Brasília)

df_correlacao_horario_voo = pd.read_csv('./input/basica2021-05.txt', error_bad_lines=False, encoding='latin', sep=';')
df_correlacao_horario_voo = df_correlacao_horario_voo.query('(nm_empresa == "GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS S.A.)") & sg_iata_origem == "GRU" & sg_iata_destino == "BSB"')
df_correlacao_horario_voo['dt_semana'] = pd.to_datetime(df_correlacao_horario_voo['dt_partida_real']).dt.day_name(locale='pt')
df_correlacao_horario_voo['hr_partida_real'] = df_correlacao_horario_voo['hr_partida_real'].str.slice(start=0, stop=2)

# Realizar o agrupamento através do horário de partida e a agregação pela MÉDIA da quantidade de RPK e ASK
df_correlacao_horario_voo = df_correlacao_horario_voo[['hr_partida_real', 'nr_voo','nr_rpk', 'nr_ask', 'dt_semana']].query('(dt_semana == "Segunda-feira")').groupby(by= 'hr_partida_real')[['hr_partida_real', 'nr_rpk', 'nr_ask']].mean().reset_index()

df_correlacao_horario_voo['hr_partida_real'] =  pd.to_numeric(df_correlacao_horario_voo['hr_partida_real'])
dfi.export(df_correlacao_horario_voo.corr(method='spearman'),"./output/imgs/df_correlacao_media_RPKASK_gru_bsb.png")
df_correlacao_horario_voo

# df_correlacao_horario_voo = df_correlacao_horario_voo[['hr_partida_real', 'nr_voo','nr_rpk', 'nr_ask', 'dt_semana']].query('(dt_semana == "Segunda-feira")').groupby(by= 'hr_partida_real')[['nr_rpk', 'nr_ask']].sum()

C:\Users\mayco\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


hr_partida_real    nr_rpk     nr_ask
0                6  100890.0  138510.00
1                8  127908.0  159030.00
2               13  134662.5  159030.00
3               14  147060.0  159030.00
4               17  134662.5  159030.00
5               22  137227.5  148983.75

In [23]:
import scipy
print("Teste de normalidade usando Shapiro-Wil para média de RPK: ", scipy.stats.shapiro(df_correlacao_horario_voo['nr_rpk']))
print("Teste de normalidade usando Shapiro-Wil para média de ASK: ", scipy.stats.shapiro(df_correlacao_horario_voo['nr_ask']))

Teste de normalidade usando Shapiro-Wil para média de RPK:  ShapiroResult(statistic=0.8431315422058105, pvalue=0.13835923373699188)
Teste de normalidade usando Shapiro-Wil para média de ASK:  ShapiroResult(statistic=0.7001284956932068, pvalue=0.006200953386723995)


In [24]:
df_correlacao_horario_voo.corr(method='spearman')

hr_partida_real    nr_rpk    nr_ask
hr_partida_real         1.000000  0.782691  0.169031
nr_rpk                  0.782691  1.000000  0.342997
nr_ask                  0.169031  0.342997  1.000000